### PR TITLE
Custom menu

### PR DESCRIPTION
**Pull request by xNatsuri**
implement custom menus to allow more flexibility with visual inventories

### CODE REVIEW

File: inv/containers.go [hunk 1]
```diff
@@ -45,3 +45,13 @@ type BarrelContainer struct{}
 func (BarrelContainer) Block() world.Block { return block.NewBarrel() }
 func (BarrelContainer) Type() int          { return protocol.ContainerBarrel }
 func (BarrelContainer) Size() int          { return 27 }
+
```
Comment (RestartFU) [inline inv/containers.go:48]: forgot to comment here

File: inv/menu.go [hunk 2]
```diff
@@ -60,13 +69,16 @@ func UpdateMenu(p *player.Player, m Menu) {
...
 
+// sendMenu sends the menu to a player.
 func sendMenu(p *player.Player, m Menu, update bool) {
 	s := player_session(p)
 
-	inv := inventory.New(m.container.Size(), func(slot int, before, after item.Stack) {})
-	inv.Handle(handler{p: p, menu: m})
-	for i, it := range m.items {
-		_ = inv.SetItem(i, it)
+	var inv *inventory.Inventory
```
Comment (RestartFU) [inline inv/menu.go:76]: this doesn't look right, shouldn't it be 

```go
inv := m.inventory

if m.custom {
// handle
}

Comment (xNatsuri) [reply inv/menu.go:76]: no if its custom theres no preset handler

Comment (xNatsuri) [reply inv/menu.go:76]: if its not custom then the handler is set

**Pull request closed without merging**